src/state/queries/notifications/feed.ts

```diff
@@ -142,20 +142,37 @@ export function* findAllPostsInQueryData(
   queryClient: QueryClient,
   uri: string,
 ): Generator<AppBskyFeedDefs.PostView, void> {
+  const atUri = new AtUri(uri)
+  const isDid = atUri.host.startsWith('did:')
+
   const queryDatas = queryClient.getQueriesData<InfiniteData<FeedPage>>({
     queryKey: [RQKEY_ROOT],
   })
   for (const [_queryKey, queryData] of queryDatas) {
     if (!queryData?.pages) {
       continue
     }
+
     for (const page of queryData?.pages) {
       for (const item of page.items) {
-        if (item.subject?.uri === uri) {
+        if (isDid && item.subject?.uri === uri) {
+          yield item.subject
+        } else if (
+          !isDid &&
+          item.subject?.author.handle === atUri.host &&
+          item.subject?.uri.endsWith(atUri.rkey)
+        ) {
           yield item.subject
         }
+
         const quotedPost = getEmbeddedPost(item.subject?.embed)
-        if (quotedPost?.uri === uri) {
+        if (isDid && quotedPost?.uri === uri) {
+          yield embedViewRecordToPostView(quotedPost)
+        } else if (
+          !isDid &&
+          quotedPost?.author.handle === atUri.host &&
+          quotedPost.uri.endsWith(atUri.rkey)
```

Comment:
This feels complicated. Ok but wondering if there are easier ways. I do think we need to pull this into a utility though cause the copy pasta is not obvious.

Reply:
Let's at least extract `isSamePostUri` or something